kafka-client/src/main/java/dev/responsive/kafka/clients/ResponsiveRestoreConsumer.java

```diff
@@ -75,12 +106,64 @@ public void seek(final TopicPartition partition, final OffsetAndMetadata offsetA
 
   @Override
   public void seekToBeginning(final Collection<TopicPartition> partitions) {
-    final Collection<TopicPartition> withUncachedPosition = seekInitial(partitions);
+    final Collection<TopicPartition> withUncachedPosition = initializeOffsets(partitions);
     if (withUncachedPosition.size() > 0) {
       // Make sure to only do this if the list of partitions without cached start positions
       // is not empty. If it's empty, KafkaConsumer will seek to the actual beginning of _all_
       // partitions.
       super.seekToBeginning(withUncachedPosition);
     }
   }
+
+  @Override
+  public void pause(final Collection<TopicPartition> partitions) {
+    final Set<TopicPartition> toPause = new HashSet<>(pausedChangelogPartitions);
+    toPause.addAll(partitions);
```

Comment:
will `pause` resume partitions that were previously paused but are not included in `partitions` parameter or is this just extra safety?